gradle/dependencies.gradle

```diff
@@ -159,6 +159,10 @@ libs += [
   jmhGeneratorAnnProcess: "org.openjdk.jmh:jmh-generator-annprocess:$versions.jmh",
   joptSimple: "net.sf.jopt-simple:jopt-simple:$versions.jopt",
   jose4j: "org.bitbucket.b_c:jose4j:$versions.jose4j",
+  // KAFKA-14109
+  // The below dependency is needed for compiling JUnit 4 tests.
+  // It can be safely removed once all of streams has moved to JUnit 5.
+  junit4: "junit:junit:$versions.junit4",
```

Comment:
This explanation is odd - compilation works currently and we don't have this dependency.

Reply:
I did some tests and discovered that `junitVintageEngine` pulls in JUnit 4. That means if we move `junitVintageEngine` back to the `testImplementation` dependencies compilation of JUnit4 tests works.

Reply:
That makes sense. There should be no need for depending on the old junit4 jars directly.

Reply:
Removed the dependency in latest revision.